crates/iota-indexer/src/handlers/committer.rs

```diff
@@ -243,3 +263,46 @@ async fn commit_checkpoints<S>(
         .thousand_transaction_avg_db_commit_latency
         .observe(elapsed * 1000.0 / tx_count as f64);
 }
+
+fn cut_out_already_indexed_object_changes(
+    already_indexed_txs: &[Vec<u8>],
+    transactions_to_commit: &[IndexedTransaction],
+    object_changes_to_commit: Vec<TransactionObjectChangesToCommit>,
+) -> Vec<TransactionObjectChangesToCommit> {
+    let already_indexed_txs: HashSet<Vec<u8>> = already_indexed_txs.iter().cloned().collect();
+
+    let already_processed_obj_versions: HashSet<(ObjectID, SequenceNumber)> =
+        transactions_to_commit
+            .iter()
+            .filter(|tx| already_indexed_txs.contains(tx.tx_digest.inner().as_slice()))
+            .flat_map(|tx| {
+                let modified_objs = tx.effects.all_changed_objects();
+                let deleted_objs = tx.effects.all_tombstones();
+                modified_objs
+                    .into_iter()
+                    .map(|o| (o.0.0, o.0.1))
```

Comment:
nit: if possible, we could add a tuple containing the two values to understand that is `o.0.0` & `o.0.1`